### PR TITLE
CVP-1597 updated search operation to raise error if query parameter i…

### PR DESCRIPTION
**Pull request by mintusah25**
Changes:

1. r4.js -> Updated buildR4SearchQuery to raise an error when the query parameter is unrecognized.
2. get_all_args.js -> Stored handling passed in the header in arguments to specify the type of search to be conducted.
3. Updated test cases.

### CODE REVIEW

**Review by imranq2 — APPROVED** [review]
just fix all the broken unit tests

**Review by imranq2 — CHANGES_REQUESTED** [review]
thanks.  please fix the unit tests.

File: src/constants.js [hunk 1]
```diff
@@ -156,4 +156,15 @@ module.exports = {
         VISIONPRESCRIPTION: 'VisionPrescription',
         CLIENT: 'Client',
     },
+    LENIENT_SEARCH_HANDLING: 'lenient',
+    STRICT_SEARCH_HANDLING: 'strict',
+    SPECIFIED_QUERY_PARAMS: [
+        '_explain', '_debug', '_validate', 'contained', '_hash_references', 'base_version', '_elements',
+        '_useAccessIndex', 'active', '_source', '_id', '_id:above', '_id:below', 'onset-date', '_lastUpdated',
+        'source', 'id', 'id:above', 'id:below', 'onset_date', '_bundle', '_sort', '_count', '_useTwoStepOptimization',
```
Comment (anmolmittal1) [inline src/constants.js:164]: Do we need id:below, id:above etc now?

File: src/operations/query/r4.js [hunk 2]
```diff
@@ -120,7 +123,11 @@ class R4SearchQueryCreator {
                 propertyObj = searchParameterQueries['Resource'][`${queryParameter}`];
             }
             if (!propertyObj) {
-                // ignore this unrecognized arg
+                // In case of an unrecognized argument while searching and handling type is strict throw an error.
+                // https://www.hl7.org/fhir/search.html#errors
+                if (args['handling'] === STRICT_SEARCH_HANDLING && specifiedQueryParams.indexOf(queryParameter) === -1) {
+                    throw new NotFoundError(`${queryParameter} is not a parameter for ${resourceType}`);
```
Comment (anmolmittal1) [inline src/operations/query/r4.js:129]: This should be a BadRequestError

File: src/operations/query/r4.js [hunk 3]
```diff
@@ -112,6 +114,7 @@ class R4SearchQueryCreator {
          */
         let totalAndSegments = [];
 
+        const specifiedQueryParams = SPECIFIED_QUERY_PARAMS;
```
Comment (anmolmittal1) [inline src/operations/query/r4.js:117]: Why is the import assigned to a local constant?

**Review by anmolmittal1 — CHANGES_REQUESTED** [review]

**Review by anmolmittal1 — CHANGES_REQUESTED** [review]

**Review by anmolmittal1 — APPROVED** [review]

**Review by 007sunny — APPROVED** [review]

**Review by imranq2 — APPROVED** [review]

**Pull request merged**